app.js

```diff
@@ -64,12 +64,12 @@ function save() {
 
 function load() {
   let parsedProducts = localStorage.getItem("productData");
-  JSON.parse(parsedProducts);
+  parsedProducts = JSON.parse(parsedProducts);
+  parsedProducts = allProducts;
 
-  for (let i = 0; i < allProducts.length; i++) {
-    let product = allProducts[i];
+  for (let i = 0; i < parsedProducts.length; i++) {
+    let product = parsedProducts[i];
     loadedProducts.push(product);
-
   }
   savedProducts = JSON.stringify(loadedProducts);
   localStorage.setItem("productData", savedProducts);
```

Comment:
`localStorage.getItem()` doesn't remove anything from local storage, it copies it, so there's no need to set `productData` again.